tools/PI/DevHome.PI/ViewModels/BarWindowViewModel.cs

```diff
@@ -363,42 +363,9 @@ private void PerfCounterHelper_PropertyChanged(object? sender, PropertyChangedEv
         }
     }
 
-    public void ExternalToolButton_Click(object sender, RoutedEventArgs e)
-    {
-        if (sender is Button clickedButton)
-        {
-            if (clickedButton.Tag is ExternalTool tool)
-            {
-                InvokeTool(tool, TargetAppData.Instance.TargetProcess?.Id, TargetAppData.Instance.HWnd);
-            }
-        }
-    }
-
     public void ManageExternalToolsButton_ExternalToolLaunchRequest(object sender, ExternalTool tool)
     {
-        InvokeTool(tool, TargetAppData.Instance.TargetProcess?.Id, TargetAppData.Instance.HWnd);
-    }
-
-    private async void InvokeTool(ExternalTool tool, int? pid, HWND hWnd)
-    {
-        try
-        {
-            var process = await tool.Invoke(pid, hWnd);
-        }
-        catch (Exception ex)
-        {
-            _log.Error(ex, "Tool launched failed");
-
-            var builder = new StringBuilder();
-            builder.AppendLine(ex.Message);
-            if (ex.InnerException is not null)
-            {
-                builder.AppendLine(ex.InnerException.Message);
-            }
-
-            var errorMessage = string.Format(CultureInfo.CurrentCulture, builder.ToString(), tool.Executable);
-            PInvoke.MessageBox(HWND.Null, errorMessage, _errorTitleText, MESSAGEBOX_STYLE.MB_ICONERROR);
-        }
+        tool.InvokeTool(null, TargetAppData.Instance.TargetProcess?.Id, TargetAppData.Instance.HWnd);
```

Comment:
Is this a different `InvokeTool` from the one in `Tool.cs`?

Reply:
Nope, one in the same. In this case, you'll get the implementation in ExternalTool.cs.

Reply:
Ah,  Okay.  I didn't see what made this particular tool and assumed it was from tool.cs.